workshops/views.py

```diff
@@ -145,51 +148,201 @@ def person_bulk_add(request):
         form = PersonBulkAddForm(request.POST, request.FILES)
         if form.is_valid():
             try:
-                persons_tasks = _upload_person_task_csv(request, request.FILES['file'])
+                persons_tasks = _upload_person_task_csv(request,
+                                                        request.FILES['file'])
             except csv.Error as e:
-                messages.add_message(request, messages.ERROR, "Error processing uploaded .CSV file: {}".format(e))
+                messages.add_message(request, messages.ERROR,
+                                     "Error processing uploaded .CSV file: {}"
+                                     .format(e))
             else:
-                for entry in persons_tasks:
-                    entry['person'].save()
-                    if entry['task']:
-                        entry['task'].person = entry['person'] # Because Django's ORM doesn't do this automatically.
-                        entry['task'].save()
-                context = {'title' : 'Process CSV File',
-                           'form': form, 
-                           'persons_tasks': persons_tasks}
-                return render(request, 'workshops/person_bulk_add_results.html', context)
+                # instead of insta-saving, put everything into session
+                # then redirect to confirmation page which in turn saves the
+                # data
+
+                request.session['bulk-add-people'] = persons_tasks
```

Comment:
Probably doesn't matter, but django's session serializer moved from pickle to json so we've got to make sure that if  in the future we add columns to the csv, the values will still be json-serializable. I'm probably just being paranoid though :)